RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -417,8 +433,50 @@ void PhotonProducer::fillPhotonCollection(edm::Event& evt,
     showerShape.hcalDepth1OverEcalBc = hcalDepth1OverEcalBc;
     showerShape.hcalDepth2OverEcalBc = hcalDepth2OverEcalBc;
     showerShape.hcalTowersBehindClusters =  TowersBehindClus;
-    newCandidate.setShowerShapeVariables ( showerShape );
-
+    /// fill extra shower shapes
+    const float spp = (!edm::isFinite(locCov[2]) ? 0. : sqrt(locCov[2]));
```

Comment:
So for sigmaIEtaIEta (ie [0] entry), its protected against. https://github.com/cms-sw/cmssw/blob/master/RecoEcal/EgammaCoreTools/interface/EcalClusterTools.h#L945-L953

For sigmaIPhiIPhi (ie [2]) looking at the dPhi calculation, it is possible for this variable to be nan
https://github.com/cms-sw/cmssw/blob/master/RecoEcal/EgammaCoreTools/interface/EcalClusterTools.h#L1165-L1182

This actually looks like there is a possibly a bug here. My reading of the code is dPhi is inf if the detID is not ecal endcap. And this may happen if any crystal in the 5x5 is not ecal endcap. I would have to run tests here to determine if this is the case. So yes, this may be guarding against a real issue. And that issue is possibly a bug which instead of fixing, somebody put a bandaid on it. 

ps: I updated my email filters to see if I'm mentioned on github better as I was missing them. Worked a charm this time, only reason I saw it :)